presto-spi/src/main/java/com/facebook/presto/spi/Page.java

```diff
@@ -121,6 +121,19 @@ public Page getRegion(int positionOffset, int length)
         return new Page(length, slicedBlocks);
     }
 
+    public Page appendColumn(Block block)
+    {
+        requireNonNull(block, "block is null");
+        if (positionCount != block.getPositionCount()) {
```

Comment:
Maybe use `checkArgument` and include both values (they have primitive overloads now that avoid boxing, so it should be free)